flow/environments/incite.py

```diff
@@ -28,24 +28,28 @@ def my_operation(job):
     """
     hostname_pattern = r'.*\.summit\.olcf\.ornl\.gov'
     template = 'summit.sh'
+    mpi_cmd_string = 'jsrun'
+    cores_per_node = 42
+    gpus_per_node = 6
 
     @staticmethod
-    def calc_num_nodes(resource_sets, cores_per_node, gpus_per_node):
+    def calc_num_nodes(resource_sets):
         cores_used = gpus_used = nodes_used = 0
         for nsets, tasks, cpus_per_task, gpus in resource_sets:
             for _ in range(nsets):
                 cores_used += tasks * cpus_per_task
                 gpus_used += gpus
-                if cores_used > cores_per_node or gpus_used > gpus_per_node:
+                if (cores_used > SummitEnvironment.cores_per_node or
+                        gpus_used > SummitEnvironment.gpus_per_node):
                     nodes_used += 1
-                    cores_used = max(0, cores_used - cores_per_node)
-                    gpus_used = max(0, gpus_used - gpus_per_node)
+                    cores_used = max(0, cores_used - SummitEnvironment.cores_per_node)
+                    gpus_used = max(0, gpus_used - SummitEnvironment.gpus_per_node)
         if cores_used > 0 or gpus_used > 0:
             nodes_used += 1
         return nodes_used
 
     @staticmethod
-    def guess_resource_sets(operation, cores_per_node, gpus_per_node):
+    def guess_resource_sets(operation):
         ntasks = max(operation.directives.get('nranks', 1), 1)
```

Comment:
Is this to prevent an error if someone give a `nranks=0`? otherwise the max is superfluous with the get.